src/sgame/sg_bot_parse.cpp

```diff
@@ -265,14 +265,19 @@ static AIValue_t botCanEvolveTo( gentity_t *self, const AIValue_t *params )
 	return AIBoxInt( BotCanEvolveToClass( self, c ) );
 }
 
-static AIValue_t humanMomentum( gentity_t*, const AIValue_t* )
+// Returns a team's momentum for use in behavior trees.
+// If the team number is invalid, debug builds will have assertion failure
+// and release builds will return the value 0, so that it's clear invalid
+// team won't have bought anything
+static AIValue_t momentum( gentity_t* self, const AIValue_t *params )
 {
-	return AIBoxInt( level.team[ TEAM_HUMANS ].momentum );
-}
-
-static AIValue_t alienMomentum( gentity_t*, const AIValue_t* )
-{
-	return AIBoxInt( level.team[ TEAM_ALIENS ].momentum );
+	int requestedTeam = AIUnBoxInt( params[ 0 ] ); //is really a team_t
+	ASSERT( G_IsPlayableTeam( requestedTeam ) );
```

Comment:
`ASSERT` should only be used for programming errors (in C++). This is an issue with user input, so it should have a log warning instead.

Reply:
I thought the ASSERT would not be compiled in release builds (like the standard one) which is why I still handled the case the line after for release builds. Guess I had not thought of searching for unv's warning facility.

Reply:
It's true, ASSERT is not compiled in release builds. But again, assertions should only be used to catch C++ programming errors within the sgame module (see e.g. https://stackoverflow.com/a/253219). The behavior tree is a user input; sgame should not crash due to a mistake in the behavior tree, regardless of build type. Nor should it silently ignore the mistake with no log message in release mode.